src/lib/formats/flopimg.h

```diff
@@ -542,6 +542,14 @@ class floppy_image
 		M2FM = 0x4D32464D  //!< "M2FM", modified modified frequency modulation
 	};
 
+	//! Sectoring
```

Comment:
Fold that into variant.  Otherwise it's going to play hell with mfi, fs_unformatted and floptool, at a minimum.

Reply:
🤔  If I fold that into variant, I'm not sure I'll be able to have the floppy drives independent from the sectoring and there won't be any real differences between this PR and Eric's. But I'll look at it some more and see if there is a way to keep the drives separate from the sectoring.

Note: I was able to specify create a new MFI file for a drive and format it inside MAME without any problems. It saved and loaded properly, when used with the drive that was configured with the correct sectoring. What about using some of the high bits of `head_count` to specify sectoring instead of using variant? 

Some of the variants seem to be redundant and have duplicate info with some of the other fields from the header in MFI, for example the DSQD. For example, SS vs DS is already specified by the `head_count` field. And QD refers to 80 track vs. 40 track drives, and number of track is also specified in `cyl_count`. The comment about QD being 80 track + DD, is not totally accurate, because those can also be formatted to SD (FM), and Heath's Hard-sectored controller only supported SD, but supported DS and 80-track drives. 

I was also confused by the distinction between SD and DD drives. Were there drive that only supported SD? Since no DD encoding required any higher rates for flex changes than the SD did, I thought that was only a controller card difference.

I saw there is an 'encoding' enum in flopimg.h, which doesn't appear in the mfi format, I wasn't sure how that info was saved in mfi files.

Reply:
@galibert I've updated it to include the sectoring info in the variant field, yet still not require duplicate floppy drives classes for each of the sectoring modes. Let me know if you see a better way to handle this. Or if there is any chance we can consider updating the MFI format to better handle sectoring.